src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs

```diff
@@ -67,18 +71,44 @@ public bool Execute(Runtime.Project project, string scriptName, Func<string, str
                 }
                 else
                 {
-                    var scriptCandiate = scriptArguments[0] + ".sh";
-                    if (File.Exists(scriptCandiate))
+                    // Special-case a script name that, perhaps with added .sh, matches an existing file.
+                    var surroundWithQuotes = false;
+                    var scriptCandidate = scriptArguments[0];
+                    if (scriptCandidate.StartsWith("\"", StringComparison.Ordinal) &&
+                        scriptCandidate.EndsWith("\"", StringComparison.Ordinal))
+                    {
+                        // Strip surrounding quotes; they were required in project.json to keep the script name
+                        // together but confuse File.Exists() e.g. "My Script", lacking ./ prefix and .sh suffix.
+                        surroundWithQuotes = true;
+                        scriptCandidate = scriptCandidate.Substring(1, scriptCandidate.Length - 2);
+                    }
+
+                    if (!scriptCandidate.EndsWith(".sh", StringComparison.Ordinal))
                     {
-                        scriptArguments[0] = scriptCandiate;
-                        scriptArguments = new[] { "/bin/bash" }.Concat(scriptArguments).ToArray();
+                        scriptCandidate = scriptCandidate + ".sh";
                     }
+
+                    if (File.Exists(Path.Combine(project.ProjectDirectory, scriptCandidate)))
+                    {
+                        // scriptCandidate may be a path relative to the project root. If so, likely will not be found
+                        // in the $PATH; add ./ to let bash know where to look.
+                        var prefix = Path.IsPathRooted(scriptCandidate) ? string.Empty : "./";
+                        var quote = surroundWithQuotes ? "\"" : string.Empty;
+                        scriptArguments[0] = $"{ quote }{ prefix }{ scriptCandidate }{ quote }";
+                    }
+
+                    // Always use /bin/bash -c in order to support redirection and so on; similar to Windows case.
+                    // Unlike Windows, must escape quotation marks within the newly-quoted string.
+                    scriptArguments = new[] { "/bin/bash", "-c", "\"" }
+                        .Concat(scriptArguments.Select(argument => argument.Replace("\"", "\\\"")))
+                        .Concat(new[] { "\"" })
+                        .ToArray();
                 }
```

Comment:
/cc @troydai I fixed the problems in this `else` block that we discussed today. E.g. code no longer checks for the script file in whatever random directory `dnu` was executing.